staging/src/k8s.io/client-go/tools/clientcmd/client_config_test.go

```diff
@@ -34,7 +34,7 @@ func TestOldMergoLib(t *testing.T) {
 	}
 	dst := T{X: "one"}
 	src := T{X: "two"}
-	mergo.Merge(&dst, &src)
+	mergo.MergeWithOverwrite(&dst, &src)
```

Comment:
this test was added to warn us when somebody revendors mergo in a new version. At least we should rename the test (I would keep it though), maybe: TestMergoSemantics. It's not about the old mergo anymore after this PR.

Reply:
Also update the comment below.

Reply:
Actually, when seeing the many changes, having some more complex smoke testing here wouldn't harm, i.e. all non-simple types like maps and slices. Then please test those before the bump of mengo and after.